libraries/core_libs/consensus/src/transaction_manager/transaction_queue.cpp

```diff
@@ -0,0 +1,65 @@
+#include "transaction_manager/transaction_queue.hpp"
+
+namespace taraxa {
+
+TransactionQueue::TransactionQueue() : prio_queue_{PriorityCompare{*this}} {}
+
+size_t TransactionQueue::size() const { return hash_queue_.size(); }
+
+bool TransactionQueue::contains(const trx_hash_t &hash) const { return hash_queue_.contains(hash); }
+
+std::shared_ptr<Transaction> TransactionQueue::get(const trx_hash_t &hash) const {
+  const auto it = hash_queue_.find(hash);
+  if (it == hash_queue_.end()) return nullptr;
+
+  return *(it->second);
+}
+
+std::vector<std::shared_ptr<Transaction>> TransactionQueue::get(uint64_t count) const {
+  if (count == 0 || prio_queue_.size() <= count) {
+    std::vector<std::shared_ptr<Transaction>> ret(prio_queue_.begin(), prio_queue_.end());
+    return ret;
+  }
+
+  std::vector<std::shared_ptr<Transaction>> ret;
+  ret.reserve(count);
+  uint16_t counter = 0;
+  for (const auto &trx : prio_queue_) {
+    if (counter == count) break;
+    ret.push_back(trx);
+    counter++;
+  }
+  return ret;
+}
+
+bool TransactionQueue::erase(const trx_hash_t &hash) {
+  // Find the hash
+  const auto it = hash_queue_.find(hash);
+  if (it == hash_queue_.end()) return false;
+
+  // Remove trx from nonce map
+  const auto nonce_it = nonce_queue_.find((*it->second)->getSender());
+  assert(nonce_it != nonce_queue_.end());
+  nonce_it->second.erase((*it->second)->getNonce());
```

Comment:
you are missing the same "assert pattern" as you used for sender, so it could happen that you dont find a bug in case there are some nonces for sender saved, but this particular tx is missing there...

Reply:
this assert was there just for development, if I am cleaning nonce_q correctly, there is no need for it anymore

Reply:
there is no way, that something can be missing anymore